### PR TITLE
Documentation for the Flannel upgrade/downgrade procedure

### PR DESCRIPTION
**Pull request by AleksandrNull**
This commit introduces documentation about flannel upgrade/downgrade procedure.

### CODE REVIEW

File: Documentation/upgrade.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Flannel upgrade/downgrade procedure
+
+There are two ways of changing flannel version in the running cluster:
+
+*1. Remove old resoursed definitions and install a new one.*
+* Pros: Cleaniest way of managing resources of the flannel deployment and no manual validation required as long as no additional resources was created by administrators/operators
```
Comment (luthermonson) [inline Documentation/upgrade.md:6]: spelling `cleanest`

Comment (rajatchopra) [inline Documentation/upgrade.md:6]: Cleanest?

File: Documentation/upgrade.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+* Cons: Massive networking outage within a cluster during the version change
+
+To follow that approach one just need to have definition of the current version of flannel and the new one. kubectl delete -f <old>.yaml and kubectl create -f <new>.yaml will do the thing
+
+*2. On the fly version*
+* Pros: Less disruptive way of changing flannel version, easier to do
+* Cons: Some version may have changes which can't be just replaced and may need resources cleanup and/or rename, manual resources comparison required
+
+To do on the fly update both old and new yaml should be compared. Resources in the new yaml with the same name or new ones can be left as is. Resources which exist in old yaml but missing in the new yaml need to be removed right after the versino change. After all of the changes validated one can just use kubectl create -f <new>.yaml and check that version changed. After that resources which are missing new applied version can be safely deleted.
```
Comment (rajatchopra) [inline Documentation/upgrade.md:15]: versino/version

File: Documentation/upgrade.md (hunk 1 above)
Comment (AleksandrNull) [reply Documentation/upgrade.md]: Thanks, fixed.

File: Documentation/upgrade.md (hunk 2 above)
Comment (AleksandrNull) [reply Documentation/upgrade.md]: Thanks, fixed!

**Review by luthermonson — APPROVED** [review]

**Pull request merged**